### PR TITLE
Fix seeds koriath

### PR DESCRIPTION
**Pull request by Koriath**
Implements a few checks to prevent calling get_gene on null, when the var "seed" was not set.
Fixes: #16800

### CODE REVIEW

**Comment by KazeEspada** [comment]
One more check on new() where it automatically deletes if it has no seed. Remove that.

**Comment by Koriath** [comment]
There's no problem on new().
It will never call a proc on a null var, because it always assigns them something valid.
What line do you mean?

**Comment by KazeEspada** [comment]
Line 21 of grown inedible

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -85,13 +85,15 @@
...
+		if(seed.get_gene(/datum/plant_gene/trait/squash))
+			squash(user)
 	..()
 
 /obj/item/weapon/reagent_containers/food/snacks/grown/throw_impact(atom/hit_atom)
-	if(!..() && seed.get_gene(/datum/plant_gene/trait/squash)) //was it caught by a mob?
-		squash(hit_atom)
+	if(seed)
+		if(!..() && seed.get_gene(/datum/plant_gene/trait/squash)) //was it caught by a mob?
+			squash(hit_atom)
```
Comment (phil235) [inline code/modules/hydroponics/grown.dm]: ..() must always be called.

**Comment by phil235** [comment]
We need to fix the underlying issue, not hide it with a bunch of if(seed) checks. Afaik growns should always have a valid seed.

**Comment by KazeEspada** [comment]
Actually, tree logs dont have a valid seed and that's intentional cause trees cant grow in hydro trays.

**Comment by phil235** [comment]
@KazeEspada Well, @Core0verload 's botany change clearly shows that all growns were needed to have a seed. See https://github.com/Core0verload/-tg-station/commit/08bdcc3024e4ba9a51d6d0d09d45a84c4be53dfd#diff-19bd37935d78687e8a0f2627c6d1396eR22

**Comment by gitgud-software** [comment]
@phil235 While I understand that you want to fix the problem in the long run, this quick patch should keep things from going out of control in the meantime. If you manage to figure out the issue you could just reverse the patch commit.

**Comment by Koriath** [comment]
In the long run, yes, fix it. In the short run though, this fix will let servers run without 4447% tick usage on the mobs subsystem.

And why is this in the mob subsystem in the first place? They're seeds, not mobs.

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (Koriath) [inline code/modules/hydroponics/grown.dm]: Done.

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -85,13 +85,17 @@
...
+	else
+		..()
```
Comment (phil235) [inline code/modules/hydroponics/grown.dm]:
```
if(!..())
	if(seed && seed.get_gene(/datum/plant_gene/trait/squash)) //was it caught by a mob?
		squash(hit_atom)
```

**Comment by phil235** [comment]
You've missed more code that needs an if(seed) check.

**Comment by Koriath** [comment]
You gotta be more specific, the part you pointed out (grown.dm, line 93 to 98) already has that check.

**Comment by phil235** [comment]
I made a PR with your changes and the ones you missed. #16818

**Comment by Cheridan** [comment]
Ok then

**Pull request closed without merging**